lib/helpers/wallet_connect_methods.dart

```diff
@@ -22,3 +24,76 @@ List<String> getLocalizedPairingMethods(
   }
   return methods;
 }
+
+String handleUnSupportedNetworkError(
+    SessionProposalErrorEvent args, AppLocalizations l10n) {
+  List<String> unSupportedNetworks = [];
+  List<String?> requiredNetworkIDs = [];
+  args.requiredNamespaces.forEach((key, value) {
+    requiredNetworkIDs.addAll(value.chains?.toList() ?? []);
+  });
+  for (var network in requiredNetworkIDs) {
+    if (network != null && network != Config.walletConnectChainId) {
+      unSupportedNetworks.add(network);
+    }
+  }
+  final title = l10n.wcErrorUnsupportedNetwork;
+  final desc = l10n.wcErrorUnsupportedNetworkDescription(
+      getUnsupportedNetworks(unSupportedNetworks));
+
+  return "$title: $desc";
+}
+
+String formatNetworkName(String network) {
+  int colonIndex = network.indexOf(':');
+  bool isEIP = network.startsWith('eip155');
+
+  // Check if the network starts with "qubic"
+  if (network.startsWith('qubic') && colonIndex > -1) {
+    String qubicEnv =
+        network.substring(colonIndex + 1); // Get the environment after "qubic:"
+    return 'Qubic ${qubicEnv[0].toUpperCase()}${qubicEnv.substring(1)}'; // Capitalize first letter of environment
+  }
+
+  // If not EIP and colon is found
+  if (!isEIP && colonIndex > -1) {
+    String name = network.substring(0, colonIndex);
+    return name[0].toUpperCase() + name.substring(1); // Capitalize first letter
+  }
+
+  // If it's an EIP network or no colon is found, return the full network
+  return network;
+}
+
+String getUnsupportedNetworks(List<String> unsupportedNetworks) {
+  if (unsupportedNetworks.length == 1) {
+    return formatNetworkName(unsupportedNetworks[0]);
+  } else {
+    final networks = unsupportedNetworks.map(formatNetworkName).toList();
+    return "$networks";
+  }
+}
+
+String? validateWalletConnectURL(String? valueCandidate, BuildContext context) {
+  final l10n = l10nOf(context);
+  const requiredLength = Config.wallectConnectUrlLength;
+  final requiredPatterns = ['wc:', 'expiryTimestamp=', 'symKey=', '@'];
+
```

Comment:
can the two first if simplified in one single if?

i.e 

if (valueCandidate.trim().length != requiredLength)

Reply:
Yes, we can convert the first three if statements actually, as they all return the same wcErrorInvalidURL